src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshCypherDecoder.java

```diff
@@ -0,0 +1,55 @@
+package uk.nhs.digital.nhsconnect.nhais.mesh;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.MeshMessage;
+import uk.nhs.digital.nhsconnect.nhais.parse.EdifactParser;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class MeshCypherDecoder {
+
+    private final EdifactParser edifactParser;
+
+    @Value("${nhais.mesh.cypherToMailbox}")
+    private final String cypherToMailbox;
+
+    @Autowired
+    public MeshCypherDecoder(EdifactParser edifactParser,
+                             @Value("${nhais.mesh.cypherToMailbox}") String cypherToMailbox) {
+        this.edifactParser = edifactParser;
+        this.cypherToMailbox = cypherToMailbox;
+    }
+
+    public String getRecipient(MeshMessage meshMessage) {
+        Map<String, String> mappings = createMappings();
+
+        String recipient = edifactParser.parse(meshMessage.getContent()).getInterchangeHeader().getRecipient();
```

Comment:
You don't need `EdifactParser` (remove it completely from this class) - pass HaTradingCode as recipient